tasks.py

```diff
@@ -37,3 +44,21 @@ def collect_over_thirty(self) -> List[Dict]:
                 self.over_thirty_records.append(record)
 
         return self.over_thirty_records
+
+
+class CompanyFieldGenerator:
+    def __init__(self, user_records, company_records) -> None:
+        self.user_records = user_records
+        self.company_records = company_records
+
+    def company_id_to_field(self) -> List[Dict]:
+        for record in self.user_records:
+            if "company_id" in record:
+                for company in self.company_records:
+                    if company["id"] == record["company_id"]:
+                        record["company"] = company
+            if "company" in record and "company_id" in record:
+                record.pop("company_id")  # only popping if there's a company record to replace the id!
```

Comment:
Why didn't you put this line right below the assignment of company?

Also, in this case it makes more sense to use ```del record['company_id']```, as the idea with pop is to use the popped value

Reply:
I was getting a keyError when I had it right under (probably because I was popping and not deleting, maybe?). I'll see if deleting instead helps, though.

Reply:
Actually, the solution on my end was to remember to break the for loop when the condition met. Thanks for making me rethink!